src/Compilers/CSharp/Portable/Symbols/PublicModel/DiscardSymbol.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class DiscardSymbol : Symbol, IDiscardSymbol
+    {
+        private readonly Symbols.DiscardSymbol _underlying;
+        private ITypeSymbol _lazyType;
+
+        public DiscardSymbol(Symbols.DiscardSymbol underlying)
+        {
+            Debug.Assert(underlying != null);
+            _underlying = underlying;
+        }
+
+        internal override CSharp.Symbol UnderlyingSymbol => _underlying;
+
+        ITypeSymbol IDiscardSymbol.Type
```

Comment:
Philosophical question: why bother explicitly implementing the interface here? I don't think we care about hiding the implementation on this type, right?

Reply:
> Philosophical question: why bother explicitly implementing the interface here? I don't think we care about hiding the implementation on this type, right?

This is my personal preference. Would like to be very explicit about how the interface is implemented, what is part of interface implementation and what is not. I very much disliked that some members were implemented explicitly before, had to spend extra time figuring out what should be moved here whereas it was very quick to locate explicit implementation syntactically and move it.

---
In reply to: [338823614](https://github.com/dotnet/roslyn/pull/39498#discussion_r338823614) [](ancestors = 338823614)